sky/serve/autoscalers.py

```diff
@@ -315,3 +369,250 @@ def _get_replica_ids_to_scale_down(num_limit: int) -> List[int]:
         if not scaling_options:
             logger.info('No scaling needed.')
         return scaling_options
+
+
+class SpotRequestRateAutoscaler(RequestRateAutoscaler,
+                                name='SPOT_REQUEST_RATE_AUTOSCALER',
+                                default=False):
+    """SpotRequestRateAutoscaler: Use spot to autoscale based on request rate.
+
+    This autoscaler uses spot instances to save cost.
```

Comment:
One reason might be the spot placer will be more costly since:
* It is currently not cost-aware, i.e. regions with different prices are chosen with equal possibility (assuming their preemption rate is the same);
* Also, using spots across regions will increase cost by nature since it means we cannot use all replicas in the cheapest zone.